### PR TITLE
MAINT: py3 remove various six usages 

### PR DESCRIPTION
**Pull request by andyfaff**
Remove six.callable imports

### CODE REVIEW

**Comment by andyfaff** [comment]
remove `six.callable, six.izip, six.reduce, six.exec_, six.integer_types`

**Comment by andyfaff** [comment]
`six.next`, `six.iteritems`, `six.itervalues`, `six.iterkeys`

**Comment by andyfaff** [comment]
Best reviewed as individual commits. I did make a couple of errors in places which were fixed in subsequent commits.

**Comment by ilayn** [comment]
Since we don't use them, I think you can clean up the `six.py` no?

**Comment by andyfaff** [comment]
I haven't cleaned up all of six.py yet, there are a few I'm not sure of.

**Comment by ev-br** [comment]
LGTM

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -38,11 +37,9 @@
...
 from ._constants import _XMAX
 
-if PY2:
-    instancemethod = types.MethodType
-else:
-    def instancemethod(func, obj, cls):
-        return types.MethodType(func, obj)
+
+def instancemethod(func, obj, cls):
+    return types.MethodType(func, obj)
```
Comment (peterbell10) [inline scipy/stats/_distn_infrastructure.py]: Would it not make more sense to remove the `instancemethod` shim completely?

Comment (andyfaff) [reply scipy/stats/_distn_infrastructure.py]: What would the change be?

Comment (andyfaff) [reply scipy/stats/_distn_infrastructure.py]: done.

**Comment by andyfaff** [comment]
There should be no more `_lib.six` usages in the codebase now.

**Comment by ev-br** [comment]
> There should be no more _lib.six usages in the codebase now. We could remove that file if wanted?

So it's dead code now, remove it I'd say. This PR or a next one, whichever is more convenient.

**Comment by ilayn** [comment]
If it is dead code +1 for removal.

**Comment by ev-br** [comment]
Thanks Andrew, all!
Is that it, then, we don't have any python 2 python code anymore? Wow.

**Comment by ev-br** [comment]
There are several instances of python 2/3 `#ifdef`s in the handwritten C extensions, but that can wait for a next PR :-)

**Comment by andyfaff** [comment]
The next one could be to remove all the relevant from __future__ statements. Pretty much every file will have one of those.

**Comment by ev-br** [comment]
Or let them be (treat them like space-after-comma pep8 violation :-))

**Comment by andyfaff** [comment]
@ev-br, those imports are typically at the top of files where they wouldn't muck up any PR's

**Comment by ev-br** [comment]
FWIW, I'm -0 on removing `__future__` imports. 
If someone does want to remove them, here's the commit which added them, might make it easy to start 

https://github.com/scipy/scipy/commit/5f4c579678e08d189f99fb7eb7c027c7d8c097ef

**Pull request merged**